File: templates/javascript/sol_part2.js
const fs = require('fs');

function main() {
    if (process.argv.length !== 3) {
        console.log('Please provide exactly one file as a parameter.');
        return;
    }

    const filePath = process.argv[2];
    try {
        const fileContent = fs.readFileSync(filePath, 'utf-8');
        // console.log(fileContent);
    } catch (error) {
        console.error(`Error reading file: ${error.message}`);
    }
}

main();

File: templates/julia/sol_part2.jl
# sol_part1.jl
function main()
    # Check if a file was provided as an argument
    if length(ARGS) != 1
        println("Please provide exactly one file as a parameter.")
        return
    end
    
    # Get the filename from the arguments
    filename = ARGS[1]
    
    # Try to open the file and read its contents
    try
        file_content = read(filename, String)
        # println(file_content)
    catch e
        println("Error reading file: $e")
    end
end

# Call the main function to run the program
main()
